From 442c7a0c8d153466912ebd41a093a435ab70c5e8 Mon Sep 17 00:00:00 2001
From: Serafin Leschke <serafin.leschke@bytesatwork.ch>
Date: Thu, 5 Sep 2019 18:14:09 +0200
Subject: [PATCH] ARM: dts: Move core bytedevkit description into a dtsi

This allows for easier adaption for customer and internal builds
---
 .../dts/stm32mp157c-bytedevkit-common.dtsi    | 319 ++++++++++++++++++
 arch/arm/boot/dts/stm32mp157c-bytedevkit.dts  | 315 +----------------
 2 files changed, 320 insertions(+), 314 deletions(-)
 create mode 100644 arch/arm/boot/dts/stm32mp157c-bytedevkit-common.dtsi

diff --git a/arch/arm/boot/dts/stm32mp157c-bytedevkit-common.dtsi b/arch/arm/boot/dts/stm32mp157c-bytedevkit-common.dtsi
new file mode 100644
index 000000000000..c12d652f40ba
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157c-bytedevkit-common.dtsi
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ *  Copyright (C) 2019 bytes at work AG - http://www.bytesatwork.ch
+ */
+
+#include "stm32mp157c-byteengine-m5.dtsi"
+#include <dt-bindings/net/ti-dp83867.h>
+
+/ {
+	model = "byteDEVKIT";
+	compatible = "batw,bytedevkit", "batw,m5", "st,stm32mp157";
+
+	aliases {
+		serial0 = &uart4;
+		ethernet0 = &ethernet0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@c0000000 {
+		reg = <0xc0000000 0x20000000>;
+	};
+
+	usb_phy_tuning: usb-phy-tuning {
+		st,hs-dc-level = <2>;
+		st,fs-rftime-tuning;
+		st,hs-rftime-reduction;
+		st,hs-current-trim = <15>;
+		st,hs-impedance-trim = <1>;
+		st,squelch-level = <3>;
+		st,hs-rx-offset = <2>;
+		st,no-lsfs-sc;
+	};
+
+	usb_vbus: regulator-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_en_vbus";
+		gpio = <&gpiob 13 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	panel_backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm2 0 2000>;
+		brightness-levels = <0 4 8 16 32 64 128 200 255>;
+		default-brightness-level = <8>;
+		power-supply = <&v3v3>;
+	};
+};
+
+&pinctrl {
+	uart4_pins_b: uart4-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('H', 13, AF8)>; /* UART4_TX */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+
+		pins2 {
+			pinmux = <STM32_PINMUX('H', 14, AF8)>; /* UART4_RX */
+			bias-disable;
+		};
+	};
+
+	uart4_idle_pins_b: uart4-idle-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('H', 13, ANALOG)>; /* UART4_TX */
+		};
+
+		pins2 {
+			pinmux = <STM32_PINMUX('H', 14, AF8)>; /* UART4_RX */
+			bias-disable;
+		};
+	};
+
+	uart4_sleep_pins_b: uart4-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('H', 13, ANALOG)>, /* UART4_TX */
+				 <STM32_PINMUX('H', 14, ANALOG)>; /* UART4_RX */
+		};
+	};
+
+	i2c3_pins: i2c3-0 {
+		pins {
+			pinmux = <STM32_PINMUX('H', 7, AF4)>, /* I2C3_SCL */
+				 <STM32_PINMUX('H', 8, AF4)>; /* I2C3_SDA */
+			bias-pull-up;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+
+	i2c3_pins_sleep: i2c3-1 {
+		pins {
+			pinmux = <STM32_PINMUX('H', 7, ANALOG)>, /* I2C3_SCL */
+				 <STM32_PINMUX('H', 8, ANALOG)>; /* I2C3_SDA */
+		};
+	};
+
+	ethernet0_rgmii_pins: rgmii-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('G', 4, AF11)>, /* ETH_RGMII_GTX_CLK */
+				 <STM32_PINMUX('B', 12, AF11)>, /* ETH_RGMII_TXD0 */
+				 <STM32_PINMUX('G', 14, AF11)>, /* ETH_RGMII_TXD1 */
+				 <STM32_PINMUX('C', 2, AF11)>, /* ETH_RGMII_TXD2 */
+				 <STM32_PINMUX('E', 2, AF11)>, /* ETH_RGMII_TXD3 */
+				 <STM32_PINMUX('G', 11, AF11)>, /* ETH_RGMII_TX_CTL */
+				 <STM32_PINMUX('A', 2, AF11)>, /* ETH_MDIO */
+				 <STM32_PINMUX('C', 1, AF11)>; /* ETH_MDC */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <2>;
+		};
+
+		pins2 {
+			pinmux = <STM32_PINMUX('C', 4, AF11)>, /* ETH_RGMII_RXD0 */
+				 <STM32_PINMUX('C', 5, AF11)>, /* ETH_RGMII_RXD1 */
+				 <STM32_PINMUX('H', 6, AF11)>, /* ETH_RGMII_RXD2 */
+				 <STM32_PINMUX('B', 1, AF11)>, /* ETH_RGMII_RXD3 */
+				 <STM32_PINMUX('A', 1, AF11)>, /* ETH_RGMII_RX_CLK */
+				 <STM32_PINMUX('A', 7, AF11)>; /* ETH_RGMII_RX_CTL */
+			bias-disable;
+		};
+	};
+
+	pwm2_pins: pwm2-0 {
+		pins {
+			pinmux = <STM32_PINMUX('G', 8, AF1)>; /* TIM2_CH1 */
+			bias-pull-down;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+	};
+
+	pwm2_sleep_pins: pwm2-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('G', 8, ANALOG)>; /* TIM2_CH1 */
+		};
+	};
+};
+
+&uart4 {
+	pinctrl-names = "default", "sleep", "idle", "no_console_suspend";
+	pinctrl-0 = <&uart4_pins_b>;
+	pinctrl-1 = <&uart4_sleep_pins_b>;
+	pinctrl-2 = <&uart4_idle_pins_b>;
+	pinctrl-3 = <&uart4_pins_b>;
+	status = "okay";
+};
+
+&sdmmc1 {
+	pinctrl-names = "default", "opendrain", "sleep";
+	pinctrl-0 = <&sdmmc1_b4_pins_a>;
+	pinctrl-1 = <&sdmmc1_b4_od_pins_a>;
+	pinctrl-2 = <&sdmmc1_b4_sleep_pins_a>;
+	broken-cd;
+	st,neg-edge;
+	bus-width = <4>;
+	vmmc-supply = <&v3v3>;
+	status = "okay";
+};
+
+&usbh_ehci {
+	phys = <&usbphyc_port0>;
+	phy-names = "usb";
+	vbus-supply = <&usb_vbus>;
+	status = "okay";
+};
+
+&usbphyc {
+	vdd3v3-supply = <&vdd_usb>;
+	status = "okay";
+};
+
+&usbphyc_port0 {
+	st,phy-tuning = <&usb_phy_tuning>;
+};
+
+/* USB OTG Port (peripheral only) */
+&usbotg_hs {
+	dr_mode = "peripheral";
+	force-b-session-valid;
+	phys = <&usbphyc_port1 0>;
+	phy-names = "usb2-phy";
+	status = "okay";
+};
+
+&usbphyc_port1 {
+	st,phy-tuning = <&usb_phy_tuning>;
+};
+
+&ethernet0 {
+	status = "okay";
+	pinctrl-0 = <&ethernet0_rgmii_pins>;
+	pinctrl-names = "default";
+	phy-mode = "rgmii";
+	phy-handle = <&phy0>;
+	st,eth_clk_sel = <1>;
+	max-speed = <1000>;
+	clock-names = "stmmaceth", "mac-clk-tx", "mac-clk-rx", "eth-ck", "syscfg-clk", "ethstp";
+	clocks = <&rcc ETHMAC>,<&rcc ETHTX>,<&rcc ETHRX>,<&rcc ETHCK_K>,<&rcc SYSCFG>,<&rcc ETHSTP>;
+
+	mdio0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "snps,dwmac-mdio";
+		phy0: ethernet-phy@0 {
+			reg = <0>;
+			ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_25_NS>;
+			ti,tx-internal-delay = <DP83867_RGMIIDCTL_250_PS>;
+			ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_8_B_NIB>;
+			ti,min-output-impedance;
+			ti,dp83867-rxctrl-strap-quirk;
+		};
+	};
+};
+
+&iwdg2 {
+	timeout-sec = <32>;
+	status = "okay";
+};
+
+&dsi {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			dsi_in: endpoint {
+				remote-endpoint = <&ltdc_ep0_out>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			dsi_out: endpoint {
+				remote-endpoint = <&dsi_panel_in>;
+			};
+		};
+	};
+
+	panel-dsi@0 {
+		compatible = "youritech,ili9806";
+		reg = <0>;
+		reset-gpios = <&gpiog 2 GPIO_ACTIVE_LOW>;
+		power-supply = <&v3v3>;
+		backlight = <&panel_backlight>;
+		status = "okay";
+
+		port {
+			dsi_panel_in: endpoint {
+				remote-endpoint = <&dsi_out>;
+			};
+		};
+	};
+};
+
+&ltdc {
+	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ltdc_ep0_out: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&dsi_in>;
+		};
+	};
+};
+
+/* Backlight PWM */
+&timers2 {
+	/* spare dmas for other usage (un-delete to enable pwm capture) */
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	status = "okay";
+
+	pwm2: pwm {
+		pinctrl-0 = <&pwm2_pins>;
+		pinctrl-1 = <&pwm2_sleep_pins>;
+		pinctrl-names = "default", "sleep";
+		#pwm-cells = <2>;
+		status = "okay";
+	};
+};
+
+&i2c3 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&i2c3_pins>;
+	pinctrl-1 = <&i2c3_pins_sleep>;
+	i2c-scl-rising-time-ns = <185>;
+	i2c-scl-falling-time-ns = <20>;
+	status = "okay";
+	/delete-property/dmas;
+	/delete-property/dma-names;
+
+	gt911@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+		interrupt-parent = <&gpiob>;
+		interrupts = <7 0>;
+
+		irq-gpios = <&gpiob 7 0>;
+		reset-gpios = <&gpiod 3 0>;
+	};
+};
+
+&rtc {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/stm32mp157c-bytedevkit.dts b/arch/arm/boot/dts/stm32mp157c-bytedevkit.dts
index e6d2f58ef3d0..e9bdf4ad8315 100644
--- a/arch/arm/boot/dts/stm32mp157c-bytedevkit.dts
+++ b/arch/arm/boot/dts/stm32mp157c-bytedevkit.dts
@@ -5,317 +5,4 @@
 
 /dts-v1/;
 
-#include "stm32mp157c-byteengine-m5.dtsi"
-#include <dt-bindings/net/ti-dp83867.h>
-
-/ {
-	model = "byteDEVKIT";
-	compatible = "batw,bytedevkit", "batw,m5", "st,stm32mp157";
-
-	aliases {
-		serial0 = &uart4;
-		ethernet0 = &ethernet0;
-	};
-
-	chosen {
-		stdout-path = "serial0:115200n8";
-	};
-
-	memory@c0000000 {
-		reg = <0xc0000000 0x20000000>;
-	};
-
-	usb_phy_tuning: usb-phy-tuning {
-		st,hs-dc-level = <2>;
-		st,fs-rftime-tuning;
-		st,hs-rftime-reduction;
-		st,hs-current-trim = <15>;
-		st,hs-impedance-trim = <1>;
-		st,squelch-level = <3>;
-		st,hs-rx-offset = <2>;
-		st,no-lsfs-sc;
-	};
-
-	usb_vbus: regulator-vbus {
-		compatible = "regulator-fixed";
-		regulator-name = "usb_en_vbus";
-		gpio = <&gpiob 13 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
-	panel_backlight: backlight {
-		compatible = "pwm-backlight";
-		pwms = <&pwm2 0 2000>;
-		brightness-levels = <0 4 8 16 32 64 128 200 255>;
-		default-brightness-level = <8>;
-		power-supply = <&v3v3>;
-	};
-};
-
-&pinctrl {
-	uart4_pins_b: uart4-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('H', 13, AF8)>; /* UART4_TX */
-			bias-disable;
-			drive-push-pull;
-			slew-rate = <0>;
-		};
-
-		pins2 {
-			pinmux = <STM32_PINMUX('H', 14, AF8)>; /* UART4_RX */
-			bias-disable;
-		};
-	};
-
-	uart4_idle_pins_b: uart4-idle-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('H', 13, ANALOG)>; /* UART4_TX */
-		};
-
-		pins2 {
-			pinmux = <STM32_PINMUX('H', 14, AF8)>; /* UART4_RX */
-			bias-disable;
-		};
-	};
-
-	uart4_sleep_pins_b: uart4-sleep-0 {
-		pins {
-			pinmux = <STM32_PINMUX('H', 13, ANALOG)>, /* UART4_TX */
-				 <STM32_PINMUX('H', 14, ANALOG)>; /* UART4_RX */
-		};
-	};
-
-	i2c3_pins: i2c3-0 {
-		pins {
-			pinmux = <STM32_PINMUX('H', 7, AF4)>, /* I2C3_SCL */
-				 <STM32_PINMUX('H', 8, AF4)>; /* I2C3_SDA */
-			bias-pull-up;
-			drive-open-drain;
-			slew-rate = <0>;
-		};
-	};
-
-	i2c3_pins_sleep: i2c3-1 {
-		pins {
-			pinmux = <STM32_PINMUX('H', 7, ANALOG)>, /* I2C3_SCL */
-				 <STM32_PINMUX('H', 8, ANALOG)>; /* I2C3_SDA */
-		};
-	};
-
-	ethernet0_rgmii_pins: rgmii-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('G', 4, AF11)>, /* ETH_RGMII_GTX_CLK */
-				 <STM32_PINMUX('B', 12, AF11)>, /* ETH_RGMII_TXD0 */
-				 <STM32_PINMUX('G', 14, AF11)>, /* ETH_RGMII_TXD1 */
-				 <STM32_PINMUX('C', 2, AF11)>, /* ETH_RGMII_TXD2 */
-				 <STM32_PINMUX('E', 2, AF11)>, /* ETH_RGMII_TXD3 */
-				 <STM32_PINMUX('G', 11, AF11)>, /* ETH_RGMII_TX_CTL */
-				 <STM32_PINMUX('A', 2, AF11)>, /* ETH_MDIO */
-				 <STM32_PINMUX('C', 1, AF11)>; /* ETH_MDC */
-			bias-disable;
-			drive-push-pull;
-			slew-rate = <2>;
-		};
-
-		pins2 {
-			pinmux = <STM32_PINMUX('C', 4, AF11)>, /* ETH_RGMII_RXD0 */
-				 <STM32_PINMUX('C', 5, AF11)>, /* ETH_RGMII_RXD1 */
-				 <STM32_PINMUX('H', 6, AF11)>, /* ETH_RGMII_RXD2 */
-				 <STM32_PINMUX('B', 1, AF11)>, /* ETH_RGMII_RXD3 */
-				 <STM32_PINMUX('A', 1, AF11)>, /* ETH_RGMII_RX_CLK */
-				 <STM32_PINMUX('A', 7, AF11)>; /* ETH_RGMII_RX_CTL */
-			bias-disable;
-		};
-	};
-
-	pwm2_pins: pwm2-0 {
-		pins {
-			pinmux = <STM32_PINMUX('G', 8, AF1)>; /* TIM2_CH1 */
-			bias-pull-down;
-			drive-push-pull;
-			slew-rate = <0>;
-		};
-	};
-
-	pwm2_sleep_pins: pwm2-sleep-0 {
-		pins {
-			pinmux = <STM32_PINMUX('G', 8, ANALOG)>; /* TIM2_CH1 */
-		};
-	};
-};
-
-&uart4 {
-	pinctrl-names = "default", "sleep", "idle", "no_console_suspend";
-	pinctrl-0 = <&uart4_pins_b>;
-	pinctrl-1 = <&uart4_sleep_pins_b>;
-	pinctrl-2 = <&uart4_idle_pins_b>;
-	pinctrl-3 = <&uart4_pins_b>;
-	status = "okay";
-};
-
-&sdmmc1 {
-	pinctrl-names = "default", "opendrain", "sleep";
-	pinctrl-0 = <&sdmmc1_b4_pins_a>;
-	pinctrl-1 = <&sdmmc1_b4_od_pins_a>;
-	pinctrl-2 = <&sdmmc1_b4_sleep_pins_a>;
-	broken-cd;
-	st,neg-edge;
-	bus-width = <4>;
-	vmmc-supply = <&v3v3>;
-	status = "okay";
-};
-
-&usbh_ehci {
-	phys = <&usbphyc_port0>;
-	phy-names = "usb";
-	vbus-supply = <&usb_vbus>;
-	status = "okay";
-};
-
-&usbphyc {
-	vdd3v3-supply = <&vdd_usb>;
-	status = "okay";
-};
-
-&usbphyc_port0 {
-	st,phy-tuning = <&usb_phy_tuning>;
-};
-
-/* USB OTG Port (peripheral only) */
-&usbotg_hs {
-	dr_mode = "peripheral";
-	force-b-session-valid;
-	phys = <&usbphyc_port1 0>;
-	phy-names = "usb2-phy";
-	status = "okay";
-};
-
-&usbphyc_port1 {
-	st,phy-tuning = <&usb_phy_tuning>;
-};
-
-&ethernet0 {
-	status = "okay";
-	pinctrl-0 = <&ethernet0_rgmii_pins>;
-	pinctrl-names = "default";
-	phy-mode = "rgmii";
-	phy-handle = <&phy0>;
-	st,eth_clk_sel = <1>;
-	max-speed = <1000>;
-	clock-names = "stmmaceth", "mac-clk-tx", "mac-clk-rx", "eth-ck", "syscfg-clk", "ethstp";
-	clocks = <&rcc ETHMAC>,<&rcc ETHTX>,<&rcc ETHRX>,<&rcc ETHCK_K>,<&rcc SYSCFG>,<&rcc ETHSTP>;
-
-	mdio0 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "snps,dwmac-mdio";
-		phy0: ethernet-phy@0 {
-			reg = <0>;
-			ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_25_NS>;
-			ti,tx-internal-delay = <DP83867_RGMIIDCTL_250_PS>;
-			ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_8_B_NIB>;
-			ti,min-output-impedance;
-			ti,dp83867-rxctrl-strap-quirk;
-		};
-	};
-};
-
-&iwdg2 {
-	timeout-sec = <32>;
-	status = "okay";
-};
-
-&dsi {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@0 {
-			reg = <0>;
-			dsi_in: endpoint {
-				remote-endpoint = <&ltdc_ep0_out>;
-			};
-		};
-
-		port@1 {
-			reg = <1>;
-			dsi_out: endpoint {
-				remote-endpoint = <&dsi_panel_in>;
-			};
-		};
-	};
-
-	panel-dsi@0 {
-		compatible = "youritech,ili9806";
-		reg = <0>;
-		reset-gpios = <&gpiog 2 GPIO_ACTIVE_LOW>;
-		power-supply = <&v3v3>;
-		backlight = <&panel_backlight>;
-		status = "okay";
-
-		port {
-			dsi_panel_in: endpoint {
-				remote-endpoint = <&dsi_out>;
-			};
-		};
-	};
-};
-
-&ltdc {
-	status = "okay";
-
-	port {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		ltdc_ep0_out: endpoint@0 {
-			reg = <0>;
-			remote-endpoint = <&dsi_in>;
-		};
-	};
-};
-
-/* Backlight PWM */
-&timers2 {
-	/* spare dmas for other usage (un-delete to enable pwm capture) */
-	/delete-property/dmas;
-	/delete-property/dma-names;
-	status = "okay";
-
-	pwm2: pwm {
-		pinctrl-0 = <&pwm2_pins>;
-		pinctrl-1 = <&pwm2_sleep_pins>;
-		pinctrl-names = "default", "sleep";
-		#pwm-cells = <2>;
-		status = "okay";
-	};
-};
-
-&i2c3 {
-	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&i2c3_pins>;
-	pinctrl-1 = <&i2c3_pins_sleep>;
-	i2c-scl-rising-time-ns = <185>;
-	i2c-scl-falling-time-ns = <20>;
-	status = "okay";
-	/delete-property/dmas;
-	/delete-property/dma-names;
-
-	gt911@5d {
-		compatible = "goodix,gt911";
-		reg = <0x5d>;
-		interrupt-parent = <&gpiob>;
-		interrupts = <7 0>;
-
-		irq-gpios = <&gpiob 7 0>;
-		reset-gpios = <&gpiod 3 0>;
-	};
-};
-
-&rtc {
-	status = "okay";
-};
+#include "stm32mp157c-bytedevkit-common.dtsi"
-- 
2.20.1

